### PR TITLE
Feature/ss58 coding update

### PR DESCRIPTION
**Pull request by ArsenyZ**
update for coding ss58 addresses with prefixes bigger than 63

### CODE REVIEW

File: IrohaCrypto/Classes/sr25519/SS58AddressFactory.h [hunk 1]
```diff
@@ -25,11 +26,11 @@ typedef NS_ENUM(NSUInteger, SNAddressFactoryError) {
 @protocol SS58AddressFactoryProtocol
 
 - (nullable NSString*)addressFromPublicKey:(id<IRPublicKeyProtocol> _Nonnull)publicKey
-                                      type:(SNAddressType)type
```
Comment (ERussel) [inline IrohaCrypto/Classes/sr25519/SS58AddressFactory.h:28]: Can you also remove SNAddressType? I think we don't need it anymore

File: IrohaCrypto/Classes/sr25519/SS58AddressFactory.m [hunk 2]
```diff
@@ -17,10 +17,27 @@
...
-                                      type:(SNAddressType)type
+                                      type:(UInt8)type
                                      error:(NSError*_Nullable*_Nullable)error {
+
+    UInt16 ident = type & 0b0011111111111111;
+
     NSMutableData *addressData = [NSMutableData data];
-    [addressData appendData:[NSData dataWithBytes:&type length:1]];
+
+    switch (ident) {
```
Comment (ERussel) [inline IrohaCrypto/Classes/sr25519/SS58AddressFactory.m:27]: Spec says that extended format encodes values between 64 and 16,383. So I think we need to use ```if (ident < 64) { ... } else { ... }```

File: IrohaCrypto/Classes/sr25519/SS58AddressFactory.m [hunk 3]
```diff
@@ -49,12 +66,28 @@ - (nullable NSString*)addressFromPublicKey:(id<IRPublicKeyProtocol> _Nonnull)pub
     return [addressData toBase58];
 }
 
+- (UInt8) decodeTypeFromData:(NSData *)addressData {
+    UInt8 first = ((uint8_t*)addressData.bytes)[0];
```
Comment (ERussel) [inline IrohaCrypto/Classes/sr25519/SS58AddressFactory.m:70]: Looking to encoding we need to check rightmost 2 bits. If 01 extended and 00 regular.

File: IrohaCrypto/Classes/sr25519/SS58AddressFactory.m [hunk 4]
```diff
@@ -49,12 +66,28 @@ - (nullable NSString*)addressFromPublicKey:(id<IRPublicKeyProtocol> _Nonnull)pub
...
+}
+
 - (nullable NSData*)accountIdFromAddress:(nonnull NSString*)address
-                                    type:(SNAddressType)type
+                                    type:(UInt8)type
                                    error:(NSError*_Nullable*_Nullable)error {
     NSData *ss58Data = [[NSData alloc] initWithBase58String:address];
 
-    if ([ss58Data length] != ADDRESS_LENGTH) {
+    if ([ss58Data length] != ADDRESS_LENGTH && [ss58Data length] != ADDRESS_LENGTH + 1) {
```
Comment (ERussel) [inline IrohaCrypto/Classes/sr25519/SS58AddressFactory.m:90]: Let't rename ADDRESS_LENGTH -> SIMPLE_ADDRESS_LENGTH and also introduce FULL_ADDRESS_LENGTH

File: IrohaCrypto/Classes/sr25519/SS58AddressFactory.m [hunk 5]
```diff
@@ -49,12 +66,28 @@ - (nullable NSString*)addressFromPublicKey:(id<IRPublicKeyProtocol> _Nonnull)pub
...
         if (error) {
             NSString *message = @"Only sr25519 account id supported";
```
Comment (ERussel) [inline IrohaCrypto/Classes/sr25519/SS58AddressFactory.m:92]: Also, please, fix this message to: ```"Address length must be %@ or %@"```

File: IrohaCrypto/Classes/sr25519/SS58AddressFactory.m [hunk 6]
```diff
@@ -112,7 +145,7 @@ - (nullable NSNumber*)typeFromAddress:(nonnull NSString*)address
                                 error:(NSError*_Nullable*_Nullable)error {
     NSData *ss58Data = [[NSData alloc] initWithBase58String:address];
 
-    if ([ss58Data length] != ADDRESS_LENGTH) {
+    if ([ss58Data length] != ADDRESS_LENGTH && [ss58Data length] != ADDRESS_LENGTH + 1) {
```
Comment (ERussel) [inline IrohaCrypto/Classes/sr25519/SS58AddressFactory.m:148]: The same as in previous function

File: IrohaCrypto/Classes/sr25519/SS58AddressFactory.m (hunk 2 above)
Comment (ArsenyZ) [reply IrohaCrypto/Classes/sr25519/SS58AddressFactory.m:27]: >(128..=255 inclusive) Reserved for future address format extensions.
so it's kind of future proofing

**Pull request merged**